astroquery/simbad/core.py

```diff
@@ -300,6 +336,50 @@ class SimbadClass(SimbadBaseQuery):
     def __init__(self):
         super().__init__()
         self._VOTABLE_FIELDS = self._VOTABLE_FIELDS.copy()
+        self._server = conf.server
+        self._tap = None
+
+    @property
+    def server(self):
+        """The Simbad mirror to use."""
+        return self._server
+
+    @server.setter
+    def server(self, server: str):
+        """Allows to switch server between Simbad mirrors.
+
+        Parameters
+        ----------
+        server : str
+            It should be one of `~astroquery.simbad.conf.servers_list`.
+        """
+        if server in conf.servers_list:
+            self._server = server
+        else:
+            raise ValueError(f"'{server}' does not correspond to a Simbad server, "
+                             f"the two existing ones are {conf.servers_list}.")
+
+    @property
+    def tap(self):
+        """A ``~pyvo.dal.tap.TAPService`` service for Simbad."""
+        tap_url = f"https://{self.server}/simbad/sim-tap"
+        # only creates a new tap instance if there are no existing one
+        # or if the server property changed since the last getter call.
+        if (not self._tap) or (self._tap.baseurl != tap_url):
+            self._tap = TAPService(baseurl=tap_url, session=self._session)
+        return self._tap
+
+    @property
+    @lru_cache(1)
+    def hardlimit(self):
+        """The maximum number of lines for Simbad's output.
+
+        This property is cached to avoid calls to simbad's capability
+        webpage each time the getter is called.
+        """
+        # replace stack of property and lru_cache by functools.cache_property when
+        # astroquery drops python 3.7 support
```

Comment:
ahh, reminder that we should indeed drop support.

Reply:
Should I keep the comment to help when the support will be dropped, or is it noise?

Reply:
please do keep comments like these. Whenever we drop support I usually go around grep for things like `3.7`, etc, so will more likely to find this for cleanup because of the comment.